### PR TITLE
design: support automatic secure boot

### PR DESCRIPTION
**Pull request by dtantsur**

### CODE REVIEW

File: design/baremetal-operator/secure-boot.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+### Non-Goals
+
+- Support for custom secure boot keys.
+- Secure boot during deployment/cleaning/inspection.
+
+## Proposal
+
+## Design Details
+
+Add a new value for `SecureBoot` enumeration: `UEFI secure boot` (constant name
```
Comment (hardys) [inline design/baremetal-operator/secure-boot.md:39]: I think this should be s/SecureBoot/BootMode ?

Comment (dtantsur) [reply design/baremetal-operator/secure-boot.md:39]: Type faster, think less :) Will fix.

**Comment by hardys** [comment]
This seems reasonable to me - adding a new BootMode value seems preferable to a new field in this case.

/cc @dhellmann

File: design/baremetal-operator/secure-boot.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+## Motivation
+
+Security-conscious deployments would like to make sure secure boot is enabled
+for their instances, so that the hardware refuses to boot kernel-level code
+that has not been signed with a known key.
+
+### Goals
+
+- API addition to enable secure boot before booting the instance (and disable
+  it on deprovisioning)
```
Comment (derekhiggins) [inline design/baremetal-operator/secure-boot.md:28]: Are there also implications here for when a node is adopted?

Comment (dtantsur) [reply design/baremetal-operator/secure-boot.md:28]: I think adopted nodes will behave the same: if we ever deprovision them, they'll have the secure boot mode removed, but only if it was requested initially (i.e. BootMode==SecureUEFI).

**Review by elfosardo — APPROVED** [review]

File: design/baremetal-operator/secure-boot.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+### Non-Goals
+
+- Support for custom secure boot keys.
+- Secure boot during deployment/cleaning/inspection.
+
+## Proposal
+
+## Design Details
+
+Add a new value for `BootMode` enumeration: `UEFI secure boot` (constant name
```
Comment (hardys) [inline design/baremetal-operator/secure-boot.md:39]: The Enum value needs updating to reflect the discussion on the implementation PR

File: design/baremetal-operator/secure-boot.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+ http://creativecommons.org/licenses/by/3.0/legalcode
+-->
+
+# Secure boot support
+
+## Status
+
+implementable
+
```
Comment (stbenjam) [inline design/baremetal-operator/secure-boot.md:13]: Can we update this to implemented, since the implementation is about to merge?

Comment (stbenjam) [reply design/baremetal-operator/secure-boot.md:13]: It's merged.

**Pull request merged**